src/t8_vtk/t8_vtk_reader.cxx

```diff
@@ -298,13 +302,108 @@ t8_vtk_iterate_cells (vtkSmartPointer < vtkDataSet > vtkGrid,
     T8_FREE (tuples);
   }
   T8_FREE (vertices);
-  return tree_id;
+  return;
+}
+
+/**
+ * Set the partition for cmesh coming from a non-distributed vtkGrid. 
+ * 
+ * \param[in, out] cmesh  The cmesh, uncommitted, partition set after call of this function 
+ * \param[in] mpirank     The mpirank of this proc.
+ * \param[in] main_proc   The rank of the main process.
+ * \param[in] num_trees   The number of trees.
+ * \param[in] dim         The dimension of the cmesh.
+ * \param[in] comm        The communicator to use. 
+ */
+static void
+t8_vtk_cmesh_partition (t8_cmesh_t cmesh, const int mpirank,
+                        const int main_proc, t8_gloidx_t num_trees, int dim,
+                        sc_MPI_Comm comm)
+{
+  t8_gloidx_t         first_tree;
+  t8_gloidx_t         last_tree;
+  if (mpirank == main_proc) {
+    first_tree = 0;
+    last_tree = num_trees - 1;
+  }
+  /* Communicate the dimension to all processes */
+  sc_MPI_Bcast (&dim, 1, sc_MPI_INT, main_proc, comm);
+  t8_cmesh_set_dimension (cmesh, dim);
+  /* Communicate the number of trees to all processes. 
+   * TODO: This probably crashes when a vtkGrid is distributed in many 
+   * files. */
+  sc_MPI_Bcast (&num_trees, 1, T8_MPI_GLOIDX, main_proc, comm);
+
+  /* Build the partition. */
+  if (mpirank < main_proc) {
+    first_tree = 0;
+    last_tree = -1;
+    t8_geometry_c      *linear_geom = t8_geometry_linear_new (dim);
+    t8_cmesh_register_geometry (cmesh, linear_geom);
+  }
+  else if (mpirank > main_proc) {
+    first_tree = num_trees;
+    last_tree = num_trees - 1;
+    t8_geometry_c      *linear_geom = t8_geometry_linear_new (dim);
+    t8_cmesh_register_geometry (cmesh, linear_geom);
+  }
+  t8_cmesh_set_partition_range (cmesh, 3, first_tree, last_tree);
+}
+
+/**
+ * Set the partition for cmesh coming from a distributed vtkGrid (like pvtu)
+ * 
+ * \param[in, out] cmesh On input a cmesh, on output a cmesh with a partition according to the number of trees read on each proc
+ * \param[in] mpirank The mpirank of this proc
+ * \param[in] mpisize The size of the mpicommunicator
+ * \param[in] num_trees The number of trees read on this proc. Must be >= 0
+ * \param[in] dim     The dimension of the cells
+ * \param[in] comm    The mpi-communicator to use. 
+ * \return            the global id of the first tree on this proc. 
+ */
+static t8_gloidx_t
+t8_vtk_distributed_partition (t8_cmesh_t cmesh, const int mpirank,
```

Comment:
I have the feeling that this function gives more or less the same results as `t8_vtk_cmesh_partition` in case of non-distributed VTK grid. Might it be a good idea to merge these two functions into one? They almost
have the same signature and suggest to do the same.
```
static void
t8_vtk_cmesh_partition (t8_cmesh_t cmesh, const int mpirank,
                        const int main_proc, t8_gloidx_t num_trees, int dim,
                        sc_MPI_Comm comm)

static t8_gloidx_t
t8_vtk_distributed_partition (t8_cmesh_t cmesh, const int mpirank,
                              const int mpisize,
                              t8_gloidx_t num_trees, int dim,
                              sc_MPI_Comm comm)
```

Reply:
That is a good idea! I updated the code, with the last push we now have only one function for it.